app/src/main/java/eu/kanade/tachiyomi/ui/player/viewer/components/CurrentChapter.kt

```diff
@@ -0,0 +1,132 @@
+        chapters: List<Chapter>? = null,
+    ) {
+        if (value != null) {
+            this.value = value
+        }
+        if (chapters != null) {
+            this.chapters = chapters
+        }
+        if (this.chapters.isEmpty()) {
+            return
+        }
+        val chapter = this.chapters.last { it.time <= (value ?: 0F) }
+        view.setComposeContent {
+            CurrentChapterComposable(
+                chapter = chapter,
+                modifier = Modifier
+                    .clickable { onClick() }
+                    .padding(end = MaterialTheme.padding.large)
+                    .wrapContentSize(Alignment.CenterStart),
+            )
+        }
+    }
+
+    @Composable
+    private fun CurrentChapterComposable(
+        chapter: Chapter,
+        modifier: Modifier = Modifier,
+    ) {
+        Box(
+            modifier = modifier
+                .clip(RoundedCornerShape(25))
+                .background(MaterialTheme.colorScheme.background.copy(alpha = 0.6F))
+                .padding(horizontal = MaterialTheme.padding.medium, vertical = MaterialTheme.padding.small),
+        ) {
+            AnimatedContent(
+                targetState = chapter,
+                transitionSpec = {
+                    if (targetState.time > initialState.time) {
+                        (slideInVertically { height -> height } + fadeIn())
+                            .togetherWith(slideOutVertically { height -> -height } + fadeOut())
+                    } else {
+                        (slideInVertically { height -> -height } + fadeIn())
+                            .togetherWith(slideOutVertically { height -> height } + fadeOut())
+                    }.using(
+                        SizeTransform(clip = false),
+                    )
+                },
+                label = "Chapter",
+            ) { currentChapter ->
+                Row {
+                    Icon(
+                        imageVector = ImageVector.vectorResource(R.drawable.ic_video_chapter_20dp),
+                        contentDescription = null,
+                        modifier = Modifier
+                            .padding(end = MaterialTheme.padding.small)
+                            .size(16.dp),
+                    )
+                    Text(
+                        text = Utils.prettyTime(currentChapter.time.toInt()),
+                        textAlign = TextAlign.Center,
+                        fontWeight = FontWeight.ExtraBold,
+                        maxLines = 1,
+                        overflow = TextOverflow.Clip,
+                        color = MaterialTheme.colorScheme.tertiary,
+                    )
+                    currentChapter.title?.let {
+                        Text(
+                            text = " • ",
+                            textAlign = TextAlign.Center,
+                            maxLines = 1,
+                            overflow = TextOverflow.Clip,
+                        )
```

Comment:
why not use `DotSeperatorText()` instead?
```suggestion
                        DotSeperatorText()
```

Reply:
cause in vertical mode, i want this to be cut off if there isn't enough space